app.js

```diff
@@ -1,87 +1,87 @@
-function _forEach(array, callback) {
-  for (let element of array) {
+Array.prototype._forEach = function(callback) {
+  for (let element of this) {
     callback(element);
   }
 }
 
-function _some(array, callback) {
-  for (let element of array) {
+Array.prototype._some = function(callback) {
+  for (let element of this) {
     if (callback(element)) {
       return true;
     }
   }
   return false;
 }
 
-function _every(array, callback) {
-  for (let element of array) {
+Array.prototype._every = function(callback) {
+  for (let element of this) {
     if (!callback(element)) {
       return false;
     }
   }
   return true;
 }
 
-function _find(array, callback) {
-  for (let element of array) {
+Array.prototype._find = function(callback) {
+  for (let element of this) {
     if (callback(element)) {
       return element;
     }
   }
   return undefined;
 }
 
-function _map(array, callback) {
+Array.prototype._map = function(callback) {
   const mappedArray = [];
   const pushToMappedArray = element => mappedArray.push(callback(element));
-  _forEach(array, pushToMappedArray);
+  this._forEach(pushToMappedArray);
   return mappedArray;
 }
 
-function _filter(array, callback) {
+Array.prototype._filter = function(callback) {
   const newArray = [];
-  _forEach(array, element => {
+  this._forEach(element => {
     if (callback(element)) {
       newArray.push(element);
     }
   });
   return newArray;
 }
 
-function _reduce(array, callback, initialValue) {
-  if (array.length !== 0) {
+Array.prototype._reduce = function(callback, initialValue) {
+  if (this.length !== 0) {
     let index;
     let accumulator;
     if (initialValue === undefined) {
       index = 1;
-      accumulator = array[0];
+      accumulator = this[0];
     } else {
       index = 0;
       accumulator = initialValue;
     }
-    for (index; index < array.length; index++) {
-      accumulator = callback(accumulator, array[index], index);
+    for (index; index < this.length; index++) {
+      accumulator = callback(accumulator, this[index], index);
     }
     return accumulator;
   }
   return undefined;
 }
 
-function _union(arrayOne, arrayTwo) {
-  const unionArray = [...arrayOne];
+Array.prototype._union = function(array) {
+  const unionArray = [...this];
   const filteredArray = _filter(
-    arrayTwo,
-    element => !_includes(arrayOne, element)
+    array,
+    element => !this._includes(element)
   );
-  _forEach(filteredArray, element => unionArray.push(element));
+  filteredArray._forEach(element => unionArray.push(element));
   return unionArray;
 }
 
-function _intersection(arrayOne, arrayTwo) {
+Array.prototype._intersection = function(array) {
   return _reduce(
-    arrayOne,
+    this,
```

Comment:
Acá el `this` no es un argumento del _reduce sino el array sobre el que se aplica el _reduce. O sea que debería ser `this._reduce( (intersection, element) => { `etc...